src/config_parser.ml

```diff
@@ -152,6 +226,159 @@ let dhcp_range end_of_directive =
   let netmask, broadcast = net_broad in
   { start_addr; end_addr; mode; netmask; broadcast; lease_time }
 
+let dhcp_host end_of_directive =
+  let until_comma =
+    scan_string () (fun () c ->
+        (* FIXME: probably be more precise in accepted characters *)
+        match c with ',' -> None | _ -> Some ())
+  in
+  let lease_time = lease_time >>| fun lease -> `Lease_time lease in
+  let id_thing =
+    string_ci "id:" *> commit
+    *> choice ~failure_msg:"Bad id thing"
+         [
+           char '*' *> return `Any_client_id;
+           (* FIXME: probably be more precise in accepted characters *)
+           ( scan false (fun is_hex -> function
+               | ',' -> None | ':' -> Some true | _ -> Some is_hex)
+           >>= fun (name, is_hex) ->
+             if is_hex then
+               (* This is not very smart *)
+               let hex_name =
+                 String.concat "" (String.split_on_char ':' name)
+               in
+               match Ohex.decode hex_name with
+               | name -> return (`Client_id name)
+               | exception Invalid_argument _ -> fail "bad hex constant"
+             else return (`Client_id name) );
+         ]
+  in
+  let net_set_thing =
+    choice [ string "net:"; string "set:" ] *> commit *> until_comma
+    >>| fun net -> `Net net
```

Comment:
I struggle with this bit -- I can see in the docstring the `set:<tag>`, but I don't quite find the `net:` string.

Reply:
It's documented in the NOTES section of the man page, very hidden. It's clearer in the source code: https://thekelleys.org.uk/gitweb/?p=dnsmasq.git;a=blob;f=src/option.c;h=92f35829ef5c12bb0f32da04d9022f0b9071825c;hb=HEAD#l4018